modules/payment/src/PaymentStorage.php

```diff
@@ -53,7 +55,13 @@ protected function doCreate(array $values) {
       $values['type'] = $payment_type->getPluginId();
     }
 
-    return parent::doCreate($values);
+    $payment = parent::doCreate($values);
+
+    // Notify other modules.
+    $event = new PaymentEvent($payment);
```

Comment:
I think we can move these 2 lines into a new private method: notifyOfPaymentCreation() (or suggest another name that fits better)

That way the comment is no longer needed and it will be easier to follow the code.

Reply:
Could be. I'm also trying to tie this to issue [2856586-add-paid-in-full-event](https://www.drupal.org/node/2856586), which depends on [2804227-add-order-getBalance](https://www.drupal.org/node/2804227) ([PR 786](https://github.com/drupalcommerce/commerce/pull/786)).

PR 786 sets the payment balance on Payment::preSave() method, which gets the $storage interface as an argument, so if we can notify general events from the EntityStorageInterface we can solve that issue too. Right now, inside the Payment object, we don't have access to the eventDispatcher...